### DISCUSSION TITLE
Chainlink Box - "VRFCoordinatorV2Mock" hit an invalid opcode while deploying.

### DISCUSSION THREAD

**Discussion by MauroPerna** [ConsenSys-archive/truffle#6123] [Truffle Boxes]
Hello everyone,

I wanted to start developing a use case using chainlink VRF recently, so I cloned the chainlink box repository.

I just downloaded it, installed the dependencies, execute docker and ganache, then ran the truffle deploy command but no matter what I do I always get the same error:

```
   Deploying 'VRFCoordinatorV2Mock'
   --------------------------------
 *** Deployment Failed ***

"VRFCoordinatorV2Mock" hit an invalid opcode while deploying. Try:
   * Verifying that your constructor params satisfy all assert conditions.
   * Verifying your constructor code doesn't access an array out of bounds.
   * Adding reason strings to your assert statements.
```

💻 OS: MacOS - Chip M2

**Comment by inex6** [comment]
I meet the same case with this.  I want to known how to fix this err. Somebody can  help me?

**Comment by FlaviuWeb3** [comment]
This error typically occurs when the contract's constructor encounters an unexpected condition or fails an assert statement.

To troubleshoot this issue, here are a few steps you can take:

Ensure dependencies: Make sure you have installed all the required dependencies correctly, including the specific versions mentioned in the Chainlink documentation. Double-check that you have the correct versions of Docker, Ganache, Truffle, and other necessary tools.

Constructor parameters: Verify that the constructor parameters you are providing for the VRFCoordinatorV2Mock contract are correct. Check the contract's constructor code and the documentation to ensure that you're passing the expected values.

Constructor code issues: Review the constructor code of the VRFCoordinatorV2Mock contract and look for any potential issues. Check for out-of-bounds array accesses or other problematic code segments.

Reason strings: As suggested in the error message, consider adding reason strings to your assert statements. Reason strings can provide more specific information about the failure, which can be helpful for debugging. For example, you can modify an assert statement like this: assert(condition, "Some reason string");

Solidity version: Ensure that you're using the correct Solidity version specified in the Chainlink documentation. It's possible that the contract you're deploying requires a specific version of Solidity to work correctly.

Gas limits: It's also possible that the deployment is failing due to exceeding the gas limit. Try increasing the gas limit when deploying the contract to see if it resolves the issue.

Check Chainlink documentation and community: Look for any specific troubleshooting steps or known issues related to deploying VRFCoordinatorV2Mock in the Chainlink documentation or community forums. Other developers may have encountered and resolved similar issues.

If none of these steps resolve the problem, it might be helpful to provide more details about your environment setup, the specific command you're running, and any additional error messages or log outputs you're receiving. With more information, we can further investigate the issue and provide more specific guidance.
